x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -414,6 +414,43 @@ class AgentPolicyService {
     return res;
   }
 
+  public async updateAllAgentPoliciesBeforeDeletingOutput(
```

Comment:
Since this API could be used for other use cases, maybe we should use a different name:
```suggestion
  public async removeOutputFromAll(
```